homer0/babel-plugin-transform-class-inject-directive
# Fix reference bug and update project

### What does this PR do?

There was a bug that when a class had only the constructor and no other method, it would throw an error. The reason was that, when there are no other methods, Babel transpiles classes to functions instead of objects, and the method in charge of adding the property for functions was doing `Object.assign` of the AST node... and I didn't know that doing it would remove the reference for `insertAfter`.

I used `Object.assign` to avoid the ESLint rule `no-param-reassign` (which makes sense), so the fix was just to use the reference and add an `eslint-disable-next-line` comment.

I also...

- Updated the project dependencies.
- Added [my ESLint plugin](http://yarnpkg.com/en/package/eslint-plugin-homer0).
- Added Node 12 to Travis.
- Updated the README.

### How should it be tested manually?

Create a class with just the constructor, add the `inject` directive and check that:

- On `master`: It will throw an error and break the process.
- On this branch: It will work.
